rebalance/rebalance.py

```diff
@@ -352,12 +353,39 @@ def liquidity_info(channel, enough_liquidity: Millisatoshi, ideal_ratio: float):
         "min": min(enough_liquidity, channel["total_msat"] / 2),
         "max": max(a_minus_b(channel["total_msat"], enough_liquidity), channel["total_msat"] / 2),
         "ideal": {}
-        }
+    }
     liquidity["ideal"]["our"] = min(max(channel["total_msat"] * ideal_ratio, liquidity["min"]), liquidity["max"])
     liquidity["ideal"]["their"] = min(max(channel["total_msat"] * (1 - ideal_ratio), liquidity["min"]), liquidity["max"])
     return liquidity
 
 
+def wait_for(success, timeout=60):
+    # cyclical lambda helper
+    # taken and modified from pyln-testing/pyln/testing/utils.py
+    start_time = time.time()
+    interval = 0.25
+    while not success() and time.time() < start_time + timeout:
+        time.sleep(interval)
+        interval *= 2
+        if interval > 5:
+            interval = 5
+    if time.time() > start_time + timeout:
+        raise ValueError("Error waiting for {}", success)
```

Comment:
Nit: `if not success():`
I think the `success` is more important here than the `timeout`.

Reply:
One question about this `wait_for_htlcs` thingy. How should the `timeout` value be? Is 60s good or do you think its more sane to use a higher value out in the wild mainnet?

Reply:
I don't know. In theory, you could ask someone else.
In practice, if you put in a log about htlc time measure, I will run this code every day, and check the logs regularly. So we may figure out the correct value soon.

Reply:
Can you re-check your remark "I think the `success` is more important here than the `timeout`."?
I force pushed the code to only have two final states (not three):
 - success()
 - timeout() and success() := False
 
 Prior to this it was:
 - success()
 - timeout() and success() := False
 - timeout() and success() := True

Does this address your concern? Also see https://github.com/ElementsProject/lightning/pull/4369 for this...

Reply:
Yes, it's a lot cleaner this way! Thank you!

Reply:
I think about changing it again (for the main repo) to make the final sleep cycle just so that it will terminate just at the time a timout would occur. But lets see what the others say. I'm not too optimistic about that change (in the main repo) as it does it can return AFTER timeout occured.

Reply:
Did a final change on that. maybe https://github.com/ElementsProject/lightning/pull/4370 is better where we do the final `time.sleep()` just with the right amount of time ;)